src/sql-parser/src/ast/defs/ddl.rs

```diff
@@ -442,39 +442,72 @@ impl AstDisplay for CsvColumns {
 }
 
 #[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub enum SourceIncludeMetadataType {
-    Key,
-    Timestamp,
-    Partition,
-    Offset,
-    Headers,
-}
-
-impl AstDisplay for SourceIncludeMetadataType {
-    fn fmt<W: fmt::Write>(&self, f: &mut AstFormatter<W>) {
-        match self {
-            SourceIncludeMetadataType::Key => f.write_str("KEY"),
-            SourceIncludeMetadataType::Timestamp => f.write_str("TIMESTAMP"),
-            SourceIncludeMetadataType::Partition => f.write_str("PARTITION"),
-            SourceIncludeMetadataType::Offset => f.write_str("OFFSET"),
-            SourceIncludeMetadataType::Headers => f.write_str("HEADERS"),
-        }
-    }
-}
-impl_display!(SourceIncludeMetadataType);
-
-#[derive(Debug, Clone, PartialEq, Eq, Hash)]
-pub struct SourceIncludeMetadata {
-    pub ty: SourceIncludeMetadataType,
-    pub alias: Option<Ident>,
+pub enum SourceIncludeMetadata {
+    Key {
```

Comment:
Have we considered keeping default column name convention for `INCLUDE HEADER` so that I can say `INCLUDE HEADER foo` and automatically map that to a column named foo? I'd expect most users to map the header to a column of the same name.

This will also keep this struct like it was before:

```rust
pub enum SourceIncludeMetadataType {
    Key,
    Timestamp,
    Partition,
    Offset,
    Headers,
    Header {
        key: String,
        use_bytes: bool,
    }
}

pub struct SourceIncludeMetadata {
    pub ty: SourceIncludeMetadataType,
    pub alias: Option<Ident>,
}
```

Reply:
Yeah, that was discussed here: https://github.com/MaterializeInc/materialize/pull/22806#discussion_r1383453626

Reply:
Yes, I don't recommend allowing the alias to be optional, for the following reason!

> The reason is that headers are very likely to not be usable directly as column names (e.g., mz-timestamp), and so users will have to double quote them to use them ("mz-timestamp"), which is guaranteed to trip people up.